proposals/0282-record-dot-syntax.rst

```diff
@@ -107,8 +118,8 @@ thus, ``f r.a.b`` parses as ``f (r.a.b)``.
 Expression     Interpretation
 ============== ===================
 ``f r.x``      means ``f (r.x)``
-``f M.n.x``    means ``f (M.n.x)``
-``f M.N.x``    means ``f (M.N.x)``
+``f M.n.x``    means ``f (M.n.x)`` (means ``f (getField @"x" M.n)``)
+``f M.N.x``    means ``f (M.N.x)`` (``M.N.x`` is a qualified name, not a record field selection)
```

Comment:
I'd like to see a note after this table saying something like this:

"M.N.x" looks ambiguous.  It could mean
* `(M.N).x`; that is, select the `x` field from the (presumably nullary) data constructor `M.N`, or
* The qualified name `M.N.x`, meaning the `x` imported from `M.N`.

The ambiguity is resolved in favour of the former.  If the latter is desired, you can still write `(M.N).x`.

Actually this entire section on precedence has two parts.
1. How to disambiguate `M.N.x` (as written above)
2. How to disambiguate `f r.x` (the main payload of this section in the original proposal.

It'd be good to lay these two points out clearly.

Reply:
- Added the note;
- Split into two points.